library/alloc/src/fmt.rs

```diff
@@ -505,23 +505,19 @@
 //! it would internally pass around this structure until it has been determined
 //! where output should go to.
 //!
-//! [`fmt::Result`]: Result
-//! [`Result`]: core::result::Result
-//! [`std::fmt::Error`]: Error
-//! [`write!`]: core::write
-//! [`write`]: core::write
-//! [`format!`]: crate::format
-//! [`to_string`]: crate::string::ToString
-//! [`writeln!`]: core::writeln
+//! [`fmt::Result`]: Result "fmt::Result"
```

Comment:
By default this won't have a tooltip, right? Why did you add one?

Reply:
In general, links generated by rustdoc do have a tooltip.

E.g. something like
```rs
/// Talking about [`Result`][std::fmt::Result] here!
pub mod m {}
```
will come with a tooltip

![Screenshot_20210906_223756](https://user-images.githubusercontent.com/3986214/132259375-864c8d00-99e8-4d82-abc2-feddc472b605.png)

Whereas (unfortunately) the following alternative
```rs
/// Talking about [`Result`] here!
/// 
/// [`Result`]: std::fmt::Result
pub mod m {}
```
creates almost the same documentation, just lacking the tooltip. That’s the general reason why I’ve converted many instances of
```markdown
[foo]: path
```
into
```markdown
[foo]: path "path"
```
and in some cases, I’ve also adjusted the tooltip slightly to make it more easy to understand. This includes the case here because `Result` is part of the prelude, a simple `"Result"` tooltip seems confusing to me, the `std::` prefix is something I’ve often avoided for tooltips OTOH (not least because the paths themselves also often are relative so some imported `std::foo` module).

<hr>

Also, as mentioned in the description of this PR, the addition of tooltips isn’t systematic and only covers places that I happened to come across. (Furthermore, the amount of tooltips added declines from earlier to later commits.)

Reply:
Ok. I think this would be better to fix systematically in rustdoc rather than adding it piecemeal to individual links. Could you open an issue and revert the tooltips you've added this PR?

Reply:
"revert" as in “add another commit to revert them” or as a “rebase / edit / force push”. Because the former could easily happen in a later PR, too; these tooltips aren't particularly hard to find (and remove) with a regex search.

Either way, the revert would only remove those cases where the path and tooltip look _identical_, right? E.g. the `fmt::Result` should better stay IMO, because I don't like a `Result` tooltip.

Reply:
Hmm, no answer. I’m assuming that an additional commit is okay, and that only the ones where tooltip and path are identical need to be removed. This is more easily accomplished by a systematic find-and-replace that by looking through all my commits here. This could, as noted, easily be a separate PR, done *after* the relevant change to rustdoc was made, but I’m including a commit here now to get this PR going.

Reply:
Sorry for the delay, I haven't had much time for rust lately.

> Either way, the revert would only remove those cases where the path and tooltip look identical, right? E.g. the fmt::Result should better stay IMO, because I don't like a Result tooltip.

Well, ideally I wouldn't want any changes, because it makes them inconsistent and it will hopefully be done by rustdoc automatically eventually ... but if you feel strongly it seems fine to keep, it's just more work for me to review.

> I’m assuming that an additional commit is okay, and that only the ones where tooltip and path are identical need to be removed.

Yup, that's fine - I don't have strong opinions about git commits.

Reply:
> I think showing the link used in the source is more clear in a way but I don't feel strongly.

The problem I see is that the link used in the source code depends on what the imports are, and the imports depends on what happens to be used in the source code. If e.g. I’m writing documentation mentioning, say, `mem::transmute`, then I’d use the path `men::transmute` (resulting in a `men::transmute` tooltip) if the module I’m in does `use core::mem` (or `use crate::mem`). But if `mem` is **only** used in docs, AFAIK I can’t just add an import for that because (I think) it’ll trigger an unused imports lint and fail CI. But it’s IMO better to *concistently* have it have the tooltip `mem::transmute` and not sometimes `core::mem::transmute` or and `crate::mem::transmute`, etc…, depending on the implementation details (i.e. imports) of the module your documentation comes from (in particular, I don’t think the `crate::` prefix is a good thing showing up in standard library tooltips).